javatests/google/registry/export/ExportConstantsTest.java

```diff
@@ -78,16 +91,40 @@ public void testBackupKinds_matchGoldenBackupKindsFile() throws Exception {
   }
 
   @Test
-  public void testReportingKinds_areSubsetOfBackupKinds() throws Exception {
-    assertThat(ExportConstants.getBackupKinds()).containsAllIn(ExportConstants.getReportingKinds());
+  public void testReportingKinds_matchGoldenReportingKindsFile() throws Exception {
+    URL goldenReportingKindsResource =
+        getResource(ExportConstantsTest.class, GOLDEN_REPORTING_KINDS_FILENAME);
+    final Pattern stripComments = Pattern.compile("\\s*#.*$");
+    List<String> goldenReportingKinds = FluentIterable
+                                   .from(Splitter.on('\n').split(
+                                       Resources.toString(goldenReportingKindsResource, UTF_8).trim()))
+                                   .transform(
+                                       new Function<String, String>() {
+                                         @Override @Nullable public String apply(@Nullable String line) {
+                                           return stripComments.matcher(line).replaceFirst("");
+                                         }})
+                                   .toList();
+    ImmutableSet<String> actualKinds = ExportConstants.getReportingKinds();
+    String updateInstructions = String.format(
+        UPDATE_REPORTING_KINDS_INSTRUCTIONS_TEMPLATE,
+        goldenReportingKindsResource.toString(),
+        Joiner.on('\n').join(actualKinds));
+    assertWithMessage(updateInstructions)
+        .that(actualKinds)
+        .containsExactlyElementsIn(goldenReportingKinds)
+        .inOrder();
   }
 
   @Test
-  public void testReportingEntityClasses_areAllBaseEntityClasses() throws Exception {
-    for (Class<? extends ImmutableObject> clazz : ExportConstants.REPORTING_ENTITY_CLASSES) {
-      assertThat(clazz.isAnnotationPresent(Entity.class))
-          .named(String.format("class %s is an @Entity", clazz.getSimpleName()))
-          .isTrue();
-    }
+  public void testReportingKinds_areSubsetOfBackupKinds() throws Exception {
+    assertThat(ExportConstants.getBackupKinds()).containsAllIn(ExportConstants.getReportingKinds());
   }
+
+  @Test public void testReportingKinds_areSubsetOfEntityClasses() throws Exception {  
```

Comment:
FWIW, this test is not as necessary as it was before, given that now we generate the reporting kinds from EntityClasses.ALL_CLASSES, so they're necessarily a subset.  (That's why there wasn't an equivalent test for the backed-up kinds.)